File: sem4/mpp/mpp-proiect-csharp-vreauladudu/persistence/repository/database/ParticipantDatabase.cs
using mpp_proiect_csharp_vreauladudu.domain;
using System;
using System.Data.SQLite;
using System.Collections.Generic;

namespace mpp_proiect_csharp_vreauladudu.repository.database
{
    public class ParticipantDatabase : AbstractDatabase<int, Participant>, ParticipantRepository
    {
        public ParticipantDatabase() : base("Participants") { }

        protected override Participant extractEntity(SQLiteDataReader result)
        {   
            int id = Convert.ToInt32(result["id"]);
            String name = Convert.ToString(result["name"]);
            int motorcycleID = Convert.ToInt32(result["mID"]);

            Participant participant = new Participant();
            participant.Id = id;
            participant.Name = name;
            participant.MotorcycleID = motorcycleID;
            return participant;
        }

        protected override SQLiteCommand findStatement(int id)
        {
            SQLiteCommand command = DatabaseUtils.GetConnection().CreateCommand();
            command.CommandText = "SELECT * FROM Participants WHERE [id] = @id";
            command.Parameters.Add(new SQLiteParameter("@id", id));
            return command;
        }

        protected override SQLiteCommand saveStatement(Participant participant)
        {
            SQLiteCommand command = DatabaseUtils.GetConnection().CreateCommand();
            command.CommandText = "INSERT INTO Participants(id, name, mID) VALUES (@id, @name, @mID)";
            command.Parameters.Add(new SQLiteParameter("@id", participant.Id));
            command.Parameters.Add(new SQLiteParameter("@name", participant.Name));
            command.Parameters.Add(new SQLiteParameter("@mID", participant.MotorcycleID));
            return command;
        }

        protected override SQLiteCommand deleteStatement(int id)
        {
            SQLiteCommand command = DatabaseUtils.GetConnection().CreateCommand();
            command.CommandText = "DELETE FROM Participants WHERE [id] = @id";
            command.Parameters.Add(new SQLiteParameter("@id", id));
            return command;
        }

        protected override SQLiteCommand updateStatement(Participant participant)
        {
            SQLiteCommand command = DatabaseUtils.GetConnection().CreateCommand();
            command.CommandText = "UPDATE Participants SET [name] = @name, [mID] = @mID WHERE [id] = @id";
            command.Parameters.Add(new SQLiteParameter("@name", participant.Name));
            command.Parameters.Add(new SQLiteParameter("@mID", participant.MotorcycleID));
            command.Parameters.Add(new SQLiteParameter("@id", participant.Id));
            return command;
        }

        public List<Participant> findByMotorcycleId(int motorcycleId)
        {
            List<Participant> participants = new List<Participant>();

            using (SQLiteCommand command = DatabaseUtils.GetConnection().CreateCommand())
            {
                command.CommandText = "SELECT * from Participants WHERE [mID] = @mID";
                command.Parameters.Add(new SQLiteParameter("@mID", motorcycleId));

                using (SQLiteDataReader result = command.ExecuteReader())
                {
                    while (result.Read())
                    {
                        participants.Add(this.extractEntity(result));
                    }
                    return participants;
                }
            }
        }
    }
}

File: sem4/mpp/mpp-proiect-java-vreauladudu/persistence/src/main/java/moto/repository/database/ParticipantDatabase.java
package moto.repository.database;

import moto.Participant;
import moto.repository.ParticipantRepository;

import java.sql.PreparedStatement;
import java.sql.ResultSet;
import java.sql.SQLException;
import java.util.ArrayList;
import java.util.List;
import java.util.Properties;

public class ParticipantDatabase extends AbstractDatabase<Integer, Participant> implements ParticipantRepository
{
    public ParticipantDatabase(Properties properties)
    {
        super(properties, "Participants");
    }

    @Override
    protected Participant extractEntity(ResultSet resultSet) throws SQLException
    {
        Integer id = resultSet.getInt("id");
        String name = resultSet.getString("name");
        Integer motorcycleID = resultSet.getInt("mID");

        Participant participant = new Participant();
        participant.setId(id);
        participant.setName(name);
        participant.setMotorcycleID(motorcycleID);
        return participant;
    }

    @Override
    protected PreparedStatement findStatement(Integer id) throws SQLException
    {
        PreparedStatement statement = DatabaseUtils.getConnection().prepareStatement("SELECT * from Participants WHERE id = ?");
        statement.setInt(1, id);
        return statement;
    }

    @Override
    protected PreparedStatement saveStatement(Participant participant) throws SQLException
    {
        PreparedStatement statement = DatabaseUtils.getConnection().prepareStatement("INSERT INTO Participants(id, name, mID) VALUES (?, ?, ?)");
        statement.setInt(1, participant.getId());
        statement.setString(2, participant.getName());
        statement.setInt(3, participant.getMotorcycleID());
        return statement;
    }

    @Override
    protected PreparedStatement deleteStatement(Integer id) throws SQLException
    {
        PreparedStatement statement = DatabaseUtils.getConnection().prepareStatement("DELETE FROM Participants WHERE id = ?");
        statement.setInt(1, id);
        return statement;
    }

    @Override
    protected PreparedStatement updateStatement(Participant participant) throws SQLException
    {
        PreparedStatement statement = DatabaseUtils.getConnection().prepareStatement("UPDATE Participants SET name = ?, mID = ? WHERE id = ?");
        statement.setString(1, participant.getName());
        statement.setInt(2, participant.getMotorcycleID());
        statement.setInt(3, participant.getId());
        return statement;
    }

    @Override
    public List<Participant> findByMotorcycleId(Integer motorcycleId) throws SQLException
    {
        List<Participant> participants = new ArrayList<>();

        PreparedStatement statement = DatabaseUtils.getConnection().prepareStatement("SELECT * from Participants WHERE mID = ?");
        statement.setInt(1, motorcycleId);
        ResultSet resultSet = statement.executeQuery();
        while (resultSet.next())
        {
            participants.add(this.extractEntity(resultSet));
        }
        return participants;
    }
}